Yoast/wordpress-seo
# Refactor TopLevelProviders and provide modal location

## Context


* I wanted to use the context providers for context rather than passing props to the containers. AdvancedSettings and FacebookEditor/TwitterEditor had such flow, to avoid conflicts when rendering both in the modal and in the metabox.

I then figured that we call `TopLevelProviders` a lot in `MetaboxFill` and `SidebarFill`, and I thought that that could be refactored. I think I was correct in that, but I want to have some extra eyes on this.

## Summary


This PR can be summarized in the following changelog entry:

* The MetaboxFill was refactored to not call TopLevelProviders around each item. Rather, TopLevelProviders is now around the MetaboxSlot.
* The SidebarFill was refactored to not call TopLevelProviders around each item. Rather, TopLevelProviders is now around the SidebarSlot.
* The AdvancedSettings, FacebookEditor, and TwitterEditor components now use location context rather than a location prop.

## Relevant technical choices:

* Providers are around the slot, rather than around each metaboxfill item.

## Test instructions

This PR can be tested by following these steps:

* Build all the things.
* Verify that the Metabox and Sidebar work as you'd expect.
* Crucial things to test are elements that are _both_ in the metabox and in the sidebar.
	* It should be possible to open both at the same time.
	* Labels in the sidebar should point to sidebar inputs, metabox labels to metabox inputs.
	* Social previews, Google preview etc, should be able to open and close without disturbing the social previews in the metabox.
* While you are at it, give it a test on premium with https://github.com/Yoast/wordpress-seo-premium/pull/2982 and merge that if it works

## UI changes

## Documentation

## Quality assurance


Fixes #
